## **BAB 2 METODE VALIDASI STATISTIKA**

# 1. UJI NORMALITAS JARQUE-BERA

## Pengertian

Uji Jarque-Bera merupakan metode statistik yang digunakan untuk mengetahui apakah suatu data berdistribusi normal atau tidak. Pengujian ini didasarkan pada nilai skewness (kemiringan) dan kurtosis (keruncingan) data. Semakin mendekati karakteristik distribusi normal, maka nilai skewness akan mendekati 0 dan nilai kurtosis akan mendekati 3.

## Hipotesis

**H₀** : Data berdistribusi normal

**H₁** : Data tidak berdistribusi normal

## Rumus

$$
JB = \frac{n}{6}\left(S^2 + \frac{(K-3)^2}{4}\right)
$$

Keterangan:

* $n$ = jumlah data
* $S$ = skewness
* $K$ = kurtosis

Rumus Skewness:

$$
S = \frac{m_3}{m_2^{3/2}}
$$

Rumus Kurtosis:

$$
K = \frac{m_4}{m_2^2}
$$

## Langkah-Langkah Algoritma

1. Menentukan data yang akan diuji.
2. Menghitung jumlah data ($n$).
3. Menghitung rata-rata data.
4. Menghitung selisih setiap data terhadap rata-rata.
5. Menghitung momen ke-2 ($m_2$).
6. Menghitung momen ke-3 ($m_3$).
7. Menghitung momen ke-4 ($m_4$).
8. Menghitung nilai skewness.
9. Menghitung nilai kurtosis.
10. Menghitung nilai Jarque-Bera.
11. Membandingkan nilai JB dengan nilai kritis Chi-Square.
12. Menentukan keputusan.

## Studi Kasus

Data yang digunakan:

```python
data = [12,15,14,10,18,16,13,17,11,14]
```

In [ ]:
data=[12,15,14,10,18,16,13,17,11,14]

In [ ]:
n=len(data)

In [ ]:
mean=sum(data)/n

In [ ]:
selisih=[]
for x in data:
  selisih.append(x-mean)

In [ ]:
m2=0
for d in selisih:
  m2+=d**2
m2=m2/n

In [ ]:
m3=0
for d in selisih:
  m3 +=d**3
m3=m3/n

In [ ]:
m4=0
for d in selisih:
  m4 +=d**4
m4=m4/n

In [ ]:
skewness=m3/(m2**1.5)

In [ ]:
kurtosis=m4/ (m2**2)

In [ ]:
jb = (n/6)*(skewness**2+((kurtosis-3)**2)/4)

In [ ]:
chi_square_critical=5.991

In [ ]:
if jb>chi_square_critical:
  keputusan="Tolak H0: data tidak berdistribusi normal"
else:
  keputusan="Gagal menolak H): data berdistribusi normal"
print("n =", n)
print("mean =", mean)
print("m2 =", m2)
print("m3 =", m3)
print("m4 =", m4)
print("skewness =", skewness)
print("kurtosis =", kurtosis)
print("JB =", jb)
print("Keputusan =", keputusan)

n = 10
mean = 14.0
m2 = 6.0
m3 = 0.0
m4 = 70.8
skewness = 0.0
kurtosis = 1.9666666666666666
JB = 0.4449074074074075
Keputusan = Gagal menolak H): data berdistribusi normal


## Intepretasi

Jika nilai $JB \leq 5.991$, maka gagal menolak H₀ sehingga data berdistribusi normal. Jika nilai $JB > 5.991$, maka H₀ ditolak sehingga data tidak berdistribusi normal.


# 2. UJI KOLMOGOROV SMIRNOV
### Pengertian Uji Kolmogorov-Smirnov (K-S)

Uji Kolmogorov-Smirnov adalah metode statistik yang digunakan untuk menguji apakah suatu sampel data mengikuti distribusi tertentu, misalnya distribusi normal. Pada uji normalitas, distribusi kumulatif data sampel dibandingkan dengan distribusi kumulatif teoritis normal. Jika perbedaan keduanya kecil, maka data dianggap mengikuti distribusi normal.

---

### Hipotesis Uji Kolmogorov-Smirnov

* H₀ (Hipotesis Nol): Data berdistribusi normal.
* H₁ (Hipotesis Alternatif): Data tidak berdistribusi normal.

---

### Langkah-Langkah Uji Kolmogorov-Smirnov

1. Menentukan data yang akan diuji

   Contoh:

   ```
   [12, 15, 14, 10, 18, 16, 13, 17, 11, 14]
   ```

2. Menghitung parameter distribusi

   * Rata-rata (Mean)
   * Standar Deviasi (Standard Deviation)

3. Membentuk distribusi normal teoritis

   Distribusi normal dibentuk berdasarkan nilai mean dan standar deviasi yang diperoleh.

4. Menghitung statistik Kolmogorov-Smirnov

   Statistik K-S (D) merupakan selisih maksimum antara fungsi distribusi kumulatif empiris data dengan distribusi kumulatif teoritis.

5. Menghitung p-value

   Nilai p-value digunakan untuk menentukan keputusan pengujian.

6. Menentukan taraf signifikansi

   Biasanya digunakan:

   ```
   α = 0,05
   ```

7. Membuat keputusan

   * Jika p-value > α, maka gagal menolak H₀ → data berdistribusi normal.
   * Jika p-value ≤ α, maka tolak H₀ → data tidak berdistribusi normal.

In [1]:
import numpy as np
from scipy.stats import kstest

def uji_kolmogorov_smirnov(data):

    mean = np.mean(data)
    std = np.std(data, ddof=1)

    statistik_ks, p_value = kstest(
        data,
        'norm',
        args=(mean, std)
    )

    alpha = 0.05

    if p_value > alpha:
        keputusan = "Gagal menolak H0: data berdistribusi normal"
    else:
        keputusan = "Tolak H0: data tidak berdistribusi normal"

    print("Jumlah Data =", len(data))
    print("Mean =", mean)
    print("Standar Deviasi =", std)
    print("Statistik KS =", statistik_ks)
    print("P-Value =", p_value)
    print("Keputusan =", keputusan)

# Data
data = [12, 15, 14, 10, 18, 16, 13, 17, 11, 14]

# Jalankan fungsi
uji_kolmogorov_smirnov(data)

Jumlah Data = 10
Mean = 14.0
Standar Deviasi = 2.581988897471611
Statistik KS = 0.09999999999999998
P-Value = 0.99963712
Keputusan = Gagal menolak H0: data berdistribusi normal


# Intepretasi
Karena nilai p-value (0.9996) lebih besar daripada α=0.05, maka gagal menolak H
0, artinya, tidak terdapat perbedaan yang signifikan antara distribusi data sampel dengan distribusi normal teoritis, sehingga data dapat dianggap mengikuti distribusi normal.

# 3. UJI HOMOGENITAS VARIANS (F VARIANS)

## Pengertian

Uji F Varians digunakan untuk mengetahui apakah dua kelompok data memiliki varians yang sama (homogen) atau berbeda secara signifikan. Uji ini sering digunakan sebagai prasyarat sebelum melakukan Uji-T maupun ANOVA.

## Hipotesis

**H₀** : Varians kedua populasi sama (homogen)

**H₁** : Varians kedua populasi tidak sama (tidak homogen)

## Rumus

$$
F = \frac{s_1^2}{s_2^2}
$$

Keterangan:

* $s_1^2$ = varians terbesar
* $s_2^2$ = varians terkecil

## Langkah-Langkah Algoritma

1. Menentukan dua kelompok data.
2. Menghitung jumlah observasi masing-masing kelompok.
3. Menghitung rata-rata masing-masing kelompok.
4. Menghitung varians kelompok pertama.
5. Menghitung varians kelompok kedua.
6. Menghitung nilai F hitung.
7. Menentukan nilai F tabel.
8. Membandingkan F hitung dengan F tabel.
9. Menentukan keputusan.

## Studi Kasus

Data kelompok pertama:

```python
data1 = [70,75,80,72,78]
```

Data kelompok kedua:

```python
data2 = [65,85,60,90,70]
```

In [ ]:
data1=[70,75,80,72,78]
data2=[65,85,60,90,70]

In [ ]:
n1=len(data1)
n2=len(data2)

In [ ]:
mean1=sum(data1)/n1
mean2=sum(data2)/n2

In [ ]:
jumlah_kuadrat1 = 0
for x in data1:
    jumlah_kuadrat1 += (x - mean1) ** 2
varians1 = jumlah_kuadrat1 / (n1 - 1)

In [ ]:
jumlah_kuadrat2 = 0
for x in data2:
  jumlah_kuadrat2 += (x - mean2) ** 2
varians2 = jumlah_kuadrat2 / (n2 - 1)

In [ ]:
if varians1 > varians2:
    f_hitung = varians1 / varians2
    df1 = n1 - 1
    df2 = n2 - 1
else:
    f_hitung = varians2 / varians1
    df1 = n2 - 1
    df2 = n1 - 1

In [ ]:
alpha = 0.05
from scipy.stats import f
f_tabel = f.ppf(1 - alpha,df1,df2)

In [ ]:
if f_hitung > f_tabel:
  keputusan = """Tolak H0: varians tidak
homogen"""
else:
  keputusan = """Gagal menolak H0: varians
homogen"""
print("n1 =", n1)
print("n2 =", n2)
print("mean1 =", mean1)
print("mean2 =", mean2)
print("varians1 =", varians1)
print("varians2 =", varians2)
print("F hitung =", f_hitung)
print("F tabel =", f_tabel)
print("Keputusan =", keputusan)

n1 = 5
n2 = 5
mean1 = 75.0
mean2 = 74.0
varians1 = 17.0
varians2 = 167.5
F hitung = 9.852941176470589
F tabel = 6.388232908695868
Keputusan = Tolak H0: varians tidak
homogen


## Intepretasi

Nilai F hitung sebesar 9,8529 lebih besar daripada F tabel sebesar 6,3882. Oleh karena itu, H₀ ditolak. Hasil ini menunjukkan bahwa varians kedua kelompok data berbeda secara signifikan atau tidak homogen.


# 4. UJI HOMOGENITAS VARIANS (LEVENE TEST)

## Pengertian

Uji Levene merupakan metode statistik yang digunakan untuk menguji apakah dua atau lebih kelompok data memiliki varians yang sama (homogen). Uji ini sering digunakan sebagai alternatif Uji F Varians karena lebih tahan (robust) terhadap pelanggaran asumsi normalitas dan keberadaan outlier.

Dalam analisis statistik, homogenitas varians merupakan salah satu asumsi penting sebelum melakukan Uji-T maupun ANOVA. Jika varians antar kelompok berbeda secara signifikan, maka hasil pengujian lanjutan dapat menjadi kurang akurat.

## Hipotesis

$$
H_0 : \sigma_1^2 = \sigma_2^2 = \cdots = \sigma_k^2
$$

Varians seluruh kelompok sama (homogen).

$$
H_1 : \text{Minimal terdapat satu kelompok yang memiliki varians berbeda}
$$

Varians tidak homogen.

## Rumus

Levene Test menghitung deviasi absolut setiap data terhadap rata-rata kelompoknya.

$$
Z_{ij} = |X_{ij} - \bar{X}_i|
$$

Keterangan:

* $X_{ij}$ = data ke-$j$ pada kelompok ke-$i$
* $\bar{X}_i$ = rata-rata kelompok ke-$i$
* $Z_{ij}$ = deviasi absolut

Setelah seluruh nilai $Z_{ij}$ diperoleh, dilakukan perhitungan statistik F menggunakan konsep ANOVA terhadap nilai deviasi absolut tersebut.

## Langkah-Langkah Algoritma

1. Menyiapkan dua atau lebih kelompok data.
2. Menghitung jumlah observasi setiap kelompok.
3. Menghitung rata-rata masing-masing kelompok.
4. Menghitung deviasi absolut setiap data terhadap rata-rata kelompoknya menggunakan rumus:

$$
Z_{ij} = |X_{ij} - \bar{X}_i|
$$

5. Menyusun seluruh nilai deviasi absolut dari setiap kelompok.
6. Menghitung rata-rata deviasi absolut tiap kelompok.
7. Menghitung rata-rata deviasi absolut keseluruhan.
8. Menghitung statistik Levene menggunakan pendekatan ANOVA.
9. Menghitung p-value atau nilai F tabel.
10. Membandingkan p-value dengan tingkat signifikansi ($\alpha = 0.05$).
11. Menentukan keputusan pengujian.

## Kriteria Keputusan

* Jika p-value > 0.05, maka gagal menolak $H_0$ (varians homogen).
* Jika p-value ≤ 0.05, maka tolak $H_0$ (varians tidak homogen).

## Studi Kasus

```python
data1 = [70, 75, 80, 72, 78]
data2 = [65, 85, 60, 90, 70]
```


In [2]:
from scipy.stats import levene

data1 = [70, 75, 80, 72, 78]
data2 = [65, 85, 60, 90, 70]

# Jumlah data
n1 = len(data1)
n2 = len(data2)

# Mean masing-masing kelompok
mean1 = sum(data1) / n1
mean2 = sum(data2) / n2

# Deviasi absolut terhadap mean
z1 = []
for x in data1:
    z1.append(abs(x - mean1))

z2 = []
for x in data2:
    z2.append(abs(x - mean2))

# Rata-rata deviasi absolut
mean_z1 = sum(z1) / len(z1)
mean_z2 = sum(z2) / len(z2)

# Uji Levene
statistik_levene, p_value = levene(data1, data2)

# Keputusan
alpha = 0.05

if p_value > alpha:
    keputusan = "Gagal menolak H0: varians homogen"
else:
    keputusan = "Tolak H0: varians tidak homogen"

# Output
print("=== UJI LEVENE ===")
print("n1 =", n1)
print("n2 =", n2)
print("mean1 =", mean1)
print("mean2 =", mean2)
print("Deviasi Absolut Kelompok 1 =", z1)
print("Deviasi Absolut Kelompok 2 =", z2)
print("Mean Deviasi Absolut 1 =", mean_z1)
print("Mean Deviasi Absolut 2 =", mean_z2)
print("Statistik Levene =", statistik_levene)
print("p-value =", p_value)
print("Keputusan =", keputusan)

=== UJI LEVENE ===
n1 = 5
n2 = 5
mean1 = 75.0
mean2 = 74.0
Deviasi Absolut Kelompok 1 = [5.0, 0.0, 5.0, 3.0, 3.0]
Deviasi Absolut Kelompok 2 = [9.0, 11.0, 14.0, 16.0, 4.0]
Mean Deviasi Absolut 1 = 3.2
Mean Deviasi Absolut 2 = 10.8
Statistik Levene = 3.4662668665667162
p-value = 0.09965604473725481
Keputusan = Gagal menolak H0: varians homogen


# Intepretasi
Karena nilai p-value (0.0997) lebih besar daripada α=0.05, maka gagal menolak H
0. Artinya, tidak terdapat perbedaan varians yang signifikan antara kelompok pertama dan kelompok kedua, sehingga kedua kelompok dapat dianggap memiliki varians yang homogen

# 5. UJI INDEPENDENSI CHI-SQUARE

## Pengertian

Uji Independensi Chi-Square digunakan untuk mengetahui apakah terdapat hubungan antara dua variabel kategorik. Pengujian dilakukan dengan membandingkan frekuensi observasi dengan frekuensi harapan.

## Hipotesis

**H₀** : Tidak terdapat hubungan antara kedua variabel (independen)

**H₁** : Terdapat hubungan antara kedua variabel (dependen)

## Rumus Frekuensi Harapan

$$
E_{ij} = \frac{(Total\ Baris_i)(Total\ Kolom_j)}{Grand\ Total}
$$

Keterangan:

* $E_{ij}$ = frekuensi harapan
* Total Baris = jumlah pada baris ke-i
* Total Kolom = jumlah pada kolom ke-j

## Rumus Statistik Chi-Square

$$
\chi^2 = \sum \frac{(O-E)^2}{E}
$$

Keterangan:

* $O$ = frekuensi observasi (Observed)
* $E$ = frekuensi harapan (Expected)

## Derajat Bebas

$$
df = (r-1)(c-1)
$$

Keterangan:

* $r$ = jumlah baris
* $c$ = jumlah kolom

## Langkah-Langkah Algoritma

1. Menentukan tabel observasi.
2. Menghitung jumlah baris dan kolom.
3. Menghitung total setiap baris.
4. Menghitung total setiap kolom.
5. Menghitung grand total.
6. Menghitung frekuensi harapan.
7. Menghitung nilai Chi-Square.
8. Menghitung derajat bebas.
9. Menentukan Chi-Square tabel.
10. Membandingkan Chi-Square hitung dengan Chi-Square tabel.
11. Menentukan keputusan.

## Studi Kasus

|           | Kopi | Matcha |
| --------- | ---- | ------ |
| Laki-laki | 30   | 20     |
| Perempuan | 15   | 35     |

Representasi Python:

```python
data = [
    [30,20],
    [15,35]
]
```



In [ ]:
observasi = [[30, 20],[15, 35]]

In [ ]:
jumlah_baris = len(observasi)
jumlah_kolom = len(observasi[0])

In [ ]:
total_baris = []
for baris in observasi:
    total_baris.append(sum(baris))

In [ ]:
total_kolom = []
for j in range(jumlah_kolom):
    total = 0
    for i in range(jumlah_baris):
        total += observasi[i][j]
    total_kolom.append(total)

In [ ]:
grand_total = sum(total_baris)

In [ ]:
harapan = []
for i in range(jumlah_baris):
    baris = []
    for j in range(jumlah_kolom):
        eij = (total_baris[i] * total_kolom[j] / grand_total)
        baris.append(eij)
    harapan.append(baris)

In [ ]:
chi_square = 0
for i in range(jumlah_baris):
    for j in range(jumlah_kolom):
        O = observasi[i][j]
        E = harapan[i][j]
        chi_square += ((O - E)**2) / E

In [ ]:
df = ((jumlah_baris - 1)*(jumlah_kolom - 1))

In [ ]:
alpha = 0.05
from scipy.stats import chi2
df = ((jumlah_baris - 1)*(jumlah_kolom - 1))
chi_tabel = chi2.ppf(1 - alpha,df)

In [ ]:
if chi_square > chi_tabel:
    keputusan = "Tolak H0: terdapat hubungan"
else:
    keputusan = """Gagal menolak H0: tidak
terdapat hubungan"""
print("Jumlah Baris =", jumlah_baris)
print("Jumlah Kolom =", jumlah_kolom)
print("Grand Total =", grand_total)
print("Harapan =", harapan)
print("Chi Square =", chi_square)
print("Chi Tabel =", chi_tabel)
print("Keputusan =", keputusan)

Jumlah Baris = 2
Jumlah Kolom = 2
Grand Total = 100
Harapan = [[22.5, 27.5], [22.5, 27.5]]
Chi Square = 9.09090909090909
Chi Tabel = 3.841458820694124
Keputusan = Tolak H0: terdapat hubungan


## Intepretasi
Nilai Chi-Square hitung sebesar 9,09 lebih besar daripada Chi-Square tabel sebesar 3,84. Oleh karena itu, H₀ ditolak. Hasil ini menunjukkan bahwa terdapat hubungan yang signifikan antara jenis kelamin dan preferensi minuman. Dengan kata lain, pilihan minuman tidak sepenuhnya independen terhadap jenis kelamin responden.

# 6. UJI INDEPENDENSI FISHER'S EXACT TEST

## Pengertian

Fisher's Exact Test merupakan metode statistik yang digunakan untuk menguji hubungan antara dua variabel kategorik pada tabel kontingensi berukuran $2 \times 2$. Uji ini sangat cocok digunakan ketika ukuran sampel relatif kecil atau ketika terdapat frekuensi harapan yang kurang dari 5 sehingga asumsi Uji Chi-Square tidak terpenuhi.

Berbeda dengan Uji Chi-Square yang menggunakan pendekatan distribusi asimtotik, Fisher's Exact Test menghitung probabilitas secara tepat (exact probability), sehingga hasilnya lebih akurat pada sampel kecil.

## Hipotesis

$$
H_0 : \text{Variabel pertama dan variabel kedua saling independen}
$$

Tidak terdapat hubungan antara kedua variabel.

$$
H_1 : \text{Variabel pertama dan variabel kedua tidak independen}
$$

Terdapat hubungan antara kedua variabel.

## Rumus

Misalkan diberikan tabel kontingensi:

|         | Kolom 1 | Kolom 2 |
| ------- | ------: | ------: |
| Baris 1 |       a |       b |
| Baris 2 |       c |       d |

Jumlah seluruh observasi:

$$
n = a+b+c+d
$$

Probabilitas tabel dihitung menggunakan distribusi hipergeometrik:

$$
P =
\frac
{(a+b)!(c+d)!(a+c)!(b+d)!}
{a!b!c!d!n!}
$$

Keterangan:

* $a,b,c,d$ = frekuensi pada masing-masing sel
* $n$ = total seluruh observasi

## Langkah-Langkah Algoritma

1. Menyusun data dalam bentuk tabel kontingensi $2 \times 2$.
2. Menghitung total setiap baris.
3. Menghitung total setiap kolom.
4. Menghitung total keseluruhan observasi.
5. Menentukan nilai $a$, $b$, $c$, dan $d$.
6. Menghitung probabilitas tabel menggunakan rumus Fisher Exact Test.
7. Menghitung probabilitas semua tabel yang sama ekstrem atau lebih ekstrem dari tabel observasi.
8. Menjumlahkan seluruh probabilitas tersebut untuk memperoleh p-value.
9. Membandingkan p-value dengan tingkat signifikansi ($\alpha = 0.05$).
10. Menentukan keputusan pengujian.

## Kriteria Keputusan

* Jika p-value > 0.05, maka gagal menolak $H_0$ (tidak terdapat hubungan).
* Jika p-value ≤ 0.05, maka tolak $H_0$ (terdapat hubungan).

## Studi Kasus

```python
data = [
    [30, 20],
    [15, 35]
]
```


In [4]:
from scipy.stats import fisher_exact

# Data observasi
#            Kopi  Matcha
# Laki-laki   30     20
# Perempuan   15     35

data = [
    [30, 20],
    [15, 35]
]

# Fisher's Exact Test
odds_ratio, p_value = fisher_exact(data)

# Taraf signifikansi
alpha = 0.05

# Keputusan
if p_value < alpha:
    keputusan = "Tolak H0: terdapat hubungan antara kedua variabel"
else:
    keputusan = "Gagal menolak H0: tidak terdapat hubungan antara kedua variabel"

# Output
print("=== UJI FISHER'S EXACT TEST ===")
print("Data Observasi =", data)
print("Odds Ratio =", odds_ratio)
print("P-Value =", p_value)
print("Alpha =", alpha)
print("Keputusan =", keputusan)

=== UJI FISHER'S EXACT TEST ===
Data Observasi = [[30, 20], [15, 35]]
Odds Ratio = 3.5
P-Value = 0.004635020180798911
Alpha = 0.05
Keputusan = Tolak H0: terdapat hubungan antara kedua variabel


## Intepretasi:

Peluang laki-laki memilih kopi sekitar 3,5 kali lebih besar dibandingkan perempuan. Namun keputusan akhir mengenai ada tidaknya hubungan tetap didasarkan pada p-value Fisher's Exact Test